### PR TITLE
CI: more versions

### LINKED ISSUE #202: TST: Expand matrix to include pytest-dev job in different OSes
As suggested by @bsipocz at https://github.com/astropy/pytest-doctestplus/pull/201#issuecomment-1581596494 .

Perhaps a weekly cron?

### LINKED ISSUE #208: ENH: Fix python 3.12 compatibility
I've tried to add a python 3.12-dev job to the matrix in #207, but am running into various problems, which are not yet clear whether to due to #200 or something else. So I'm opening this issue as a blocker for the first new release under the new org.

### PR DESCRIPTION
**Pull request by bsipocz**
This is to close #202 

I'll let CI run, and then will modify one of the devtests to run with python 3.12, which I expect may cause some failures.

Close #208

### CODE REVIEW

**Comment by bsipocz** [comment]
As at the end none of the changes was required for 3.12 compatibility, I don't add a changelog for this PR, but will do a one liner `- Changing GitHub organization` changelog, and do a 1.0.0 once this is in.

**Review by jarrodmillman — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -20,7 +23,14 @@ deps =
     pytest61: pytest==6.1.*
     pytest62: pytest==6.2.*
     pytest70: pytest==7.0.*
+    pytest71: pytest==7.1.*
+    pytest72: pytest==7.2.*
+    pytest73: pytest==7.3.*
+    pytest74: pytest==7.4.*
     pytestdev: git+https://github.com/pytest-dev/pytest#egg=pytest
+    # We keep seeing pytest related issues with the latest release and python 3.12, keep using the dev version until it's fixed upstream
```
Comment (pllim) [inline tox.ini:31]: Oh, interesting. Should we cross-link those issues here?

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by pllim** [comment]
I dunno if you want to cross-link issues or not, so I'll let you merge.

File: tox.ini (hunk 1 above)
Comment (bsipocz) [reply tox.ini:31]: Not sure how real those issues are, I would just leave this for now as is and open a reminder that we need to check what our minimum pytest requirement should be for 3.12. (e.g. I don't think it makes sense just yet to add any limits to setup.cfg, would you disagree?)

Comment (pllim) [reply tox.ini:31]: That is fine by me!

**Comment by pllim** [comment]
p.s. Should we enable "Squash and Merge" button in this repo?

**Comment by bsipocz** [comment]
Nope, with the power vested in me, I turned that off as I don't like that one (still like to run consistency checks, and don't want to rewrite that stuff that does it). So I would rather enforce people to make reasonable commits, rebase and squash when needed in the PR, after all contributors here are expected to know enough git. :)

**Pull request merged**